File: basic/src/main/java/top/non/late/pavilion/basic/java/designs/builder/code/SpringCloudBookBuilder.java
package top.non.late.pavilion.basic.java.designs.builder.code;

/**
 * Created by nonlatepavilion@outlook.com on 2017/9/7.
 * Desc:
 */
public class SpringCloudBookBuilder implements BookBuilder {
    private BookProduct product;

    public SpringCloudBookBuilder() {
        this.product = new BookProduct();
    }


    @Override
    public void bookName() {
        this.product.setBookName("Spring Cloud微服务实战");
    }

    @Override
    public void author() {
        this.product.setAuthor("翟永超");
    }

    @Override
    public void isbn() {
        this.product.setIsbn("9787121313011");
    }

    @Override
    public BookProduct build() {
        return product;
    }
}

File: basic/src/main/java/top/non/late/pavilion/basic/kotlin/designs/builder/SpringCloudBookBuilder.kt
package top.non.late.pavilion.basic.kotlin.designs.builder

/**
 * Created by nonlatepavilion@outlook.com on 2017/9/25.
 * Desc: 
 */
class SpringCloudBookBuilder : BookBuilder {
    private val product = BookProduct()
    override fun bookName() {
        product.bookName = "Spring Cloud微服务实战"
    }

    override fun author() {
        product.author = "翟永超"
    }

    override fun isbn() {
        product.isbn = "9787121313011"
    }

    override fun build(): BookProduct {
        return product
    }

}